src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriterAsync.cs

```diff
@@ -1841,6 +1846,19 @@ protected internal override async Task WriteStartAttributeAsync(string? prefix,
             await base.WriteStartAttributeAsync(prefix, localName, ns).ConfigureAwait(false);
         }
 
+        // Same as base class, plus possible indentation.
+        internal override async Task WriteStartNamespaceDeclarationAsync(string prefix)
+        {
+            CheckAsyncCall();
+            // Add indentation
+            if (_newLineOnAttributes)
+            {
+                await WriteIndentAsync().ConfigureAwait(false);
+            }
+
+            await base.WriteStartNamespaceDeclarationAsync(prefix).ConfigureAwait(false);
```

Comment:
just curious, why not move this logic to base?

Reply:
Well, I gues you know more about the design decisions of this stuff, but all the indentation is only done in these `XxxRawTextWriterIndent` classes. I assume to save some bytes when no indentation is needed.

Reply:
Design was made long time before I joined the team 😄